File: 2020/src/AoC2020/Day1.cs
using System;
using System.Collections.Generic;

namespace AoC2020
{
    public static class Day1
    {
        public static int CalculatePart1(List<string> puzzleInput)
        {
            for (int i = 0; i < puzzleInput.Count; i++) 
            {
                for (int j = 1; j < puzzleInput.Count; j++)
                {
                    if (int.Parse(puzzleInput[i]) + int.Parse(puzzleInput[j]) == 2020)
                    {
                        return int.Parse(puzzleInput[i]) * int.Parse(puzzleInput[j]);
                    }
                }
            }

            return -1;
        }

        public static int CalculatePart2(List<string> puzzleInput)
        {
            for (int i = 0; i < puzzleInput.Count; i++) 
            {
                for (int j = 1; j < puzzleInput.Count; j++)
                {
                    for (int k = 2; k < puzzleInput.Count; k++)
                    {
                        if (int.Parse(puzzleInput[i]) + int.Parse(puzzleInput[j]) + int.Parse(puzzleInput[k]) == 2020)
                        {
                            return int.Parse(puzzleInput[i]) * int.Parse(puzzleInput[j]) * int.Parse(puzzleInput[k]);
                        }
                    }
                }
            }

            return -1;
        }
    }
}

File: 2021/src/com/adventofcode/Day1.java
package com.adventofcode;

import java.util.List;

public class Day1 {
    public static int calculatePart1(List<String> puzzleInput) {
        int increaseCount = 0;

        for (int i = 1; i < puzzleInput.size(); i++) {
            if (Integer.parseInt(puzzleInput.get(i)) > Integer.parseInt(puzzleInput.get(i - 1))) {
                increaseCount++;
            }
        }

        return increaseCount;
    }

    public static int calculatePart2(List<String> puzzleInput) {
        int increaseCount = 0;

        for (int i = 0; i < puzzleInput.size(); i++) {
            if (i + 3 > puzzleInput.size() - 1) {
                break;
            }

            int firstSlidingWindowSum = 0;
            for (int j = 0; j < 3; j++) {
                firstSlidingWindowSum += Integer.parseInt(puzzleInput.get(i + j));
            }

            int secondSlidingWindowSum = 0;
            for (int j = 0; j < 3; j++) {
                secondSlidingWindowSum += Integer.parseInt(puzzleInput.get(i + 1 + j));
            }

            if (secondSlidingWindowSum > firstSlidingWindowSum) {
                increaseCount++;
            }
        }

        return increaseCount;
    }
}
